File: shopy-api/Models/PlatformSettings.cs
namespace shopy_api.Models;

/// <summary>
/// Baris tunggal (singleton, <see cref="Id"/> selalu 1) berisi parameter bisnis
/// platform yang bisa diubah admin tanpa redeploy (TASKSELLER.md Fase 9) —
/// menggantikan pembacaan langsung <c>IConfiguration:Platform:*</c> yang
/// sebelumnya statis dari <c>appsettings.json</c>.
/// </summary>
public class PlatformSettings
{
    public int Id { get; set; } = 1;

    public decimal CommissionPercent { get; set; }
    public decimal WithdrawalAdminFee { get; set; }
    public decimal MinWithdrawal { get; set; }
    public int MaxWithdrawalsPerDay { get; set; }
    public int AutoCancelHours { get; set; }
    public int AutoCompleteDays { get; set; }
    public int LowStockThreshold { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: shopy-admin/lib/models/settings/platform_settings.dart
/// Sesuai `PlatformSettingsDto` di backend.
class PlatformSettings {
  final double commissionPercent;
  final int withdrawalAdminFee;
  final int minWithdrawal;
  final int maxWithdrawalsPerDay;
  final int autoCancelHours;
  final int autoCompleteDays;
  final int lowStockThreshold;
  final DateTime updatedAt;

  const PlatformSettings({
    required this.commissionPercent,
    required this.withdrawalAdminFee,
    required this.minWithdrawal,
    required this.maxWithdrawalsPerDay,
    required this.autoCancelHours,
    required this.autoCompleteDays,
    required this.lowStockThreshold,
    required this.updatedAt,
  });

  factory PlatformSettings.fromJson(Map<String, dynamic> json) {
    return PlatformSettings(
      commissionPercent: (json['commissionPercent'] as num).toDouble(),
      withdrawalAdminFee: (json['withdrawalAdminFee'] as num).round(),
      minWithdrawal: (json['minWithdrawal'] as num).round(),
      maxWithdrawalsPerDay: json['maxWithdrawalsPerDay'] as int,
      autoCancelHours: json['autoCancelHours'] as int,
      autoCompleteDays: json['autoCompleteDays'] as int,
      lowStockThreshold: json['lowStockThreshold'] as int,
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
}
